src/main/kotlin/org/rust/lang/core/grammar/rust.bnf

```diff
@@ -662,8 +663,7 @@ type_qual_path ::= << qualified_path view_path_prefix >>
 
 private types ::= type !'=' (',' type !'=')*
 
-private fn_anon_params ::= '(' anon_parameter anon_params_allow_variadic_tail ')'
-                         | '(' ')'
+bare_fn_params ::= '(' [ anon_parameter anon_params_allow_variadic_tail ] ')' { pin = 1 }
```

Comment:
I would suggest renaming this to `fn_type_params`. 

`bare_fn` is from the days of youre, when closure_type and proc_type were around: https://github.com/rust-lang/rust/commit/37afc528eacf4a0a874f8e97f6e7d4d7b63e8ac6